### PR TITLE
quiche: build quic http libraries

### PR DESCRIPTION
**Pull request by danzh2010**
Add build targets for all files under quiche/quic/core/http/. These libraries brings in QUIC http interfaces. As Envoy doesn't have GURL which quiche/quic/core/http/spdy_server_push_utils.cc depends on, it's replaced by //source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc as the definition of all the functions defined in spdy_server_push_utils.h. These envoy version function definitions does nothing but immediately fail if reached, because Envoy doesn't support server push.
 
Update quiche tar ball: 2a930469533c3b541443488a629fe25cd8ff53d0

Rename quic_platform_mem_slice_storage_lib to quic_platform_mem_slice_storage to follow platform impl's naming convention.

Testing: no test is needed as long as the code compile
Part of #2557

### CODE REVIEW

File: source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// quiche/quic/core/http/spdy_server_push_utils.cc which depends on GURL.
+// Since Envoy doesn't support server push, these functions shouldn't be
+// executed at all.
+
+using spdy::SpdyHeaderBlock;
+
+namespace quic {
+
+// static
+std::string SpdyServerPushUtils::GetPromisedUrlFromHeaders(const SpdyHeaderBlock& /*headers*/){
```
Comment (wu-bin) [inline source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc]: nit: space before '{' and new line before '}'. Same for PromisedUrlIsValid.

Comment (wu-bin) [inline source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc]: Should we use `NOT_IMPLEMENTED_GCOVR_EXCL_LINE` PANIC instead throughout this file?

Comment (danzh2010) [reply source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc]: Yeah, NOT_IMPLEMENTED_GCOVR_EXCL_LINE seems more suitable here. Done

Comment (danzh2010) [reply source/extensions/quic_listeners/quiche/spdy_server_push_utils_for_envoy.cc]: done.

**Pull request merged**